internal/spentcalories/spentCalories.go

```diff
@@ -1,18 +1,52 @@
 package spentcalories
 
 import (
+	"fmt"
+	"strconv"
+	"strings"
 	"time"
 )
 
 // Основные константы, необходимые для расчетов.
 const (
-	lenStep = 0.65 // средняя длина шага.
-	mInKm   = 1000 // количество метров в километре.
-	minInH  = 60   // количество минут в часе.
+	lenStep   = 0.65  // средняя длина шага.
+	mInKm     = 1000  // количество метров в километре.
+	minInH    = 60    // количество минут в часе.
+	kmhInMsec = 0.278 // коэффициент для преобразования км/ч в м/с.
+	cmInM     = 100   // количество сантиметров в метре.
 )
 
+//Функция принимает строку с данными формата "3456,Ходьба,3h00m", которая содержит количество шагов, вид активности и продолжительность активности. Функция возвращает четыре значения:
+//int — количество шагов.
+//string — вид активности.
+//time.Duration — продолжительность активности.
+//error — ошибку, если что-то пошло не так.
+
 func parseTraining(data string) (int, string, time.Duration, error) {
-	// ваш код ниже
+	var period time.Duration
+	//	strData := strings.Split(data, ",")
+	strData := strings.Split(data, ",")
+	if len(strData) != 3 {
+		return 0, "0", period, fmt.Errorf("[parseTraining] неверный формат строки: %s", data)
+	}
+	step, err := strconv.Atoi(strData[0])
+	if err != nil {
+		return 0, "0", period, fmt.Errorf("[parseTraining] неверный формат количества шагов: %s", strData[0])
```

Comment:
Если ошибка, то отправляем пустой ответ
```suggestion
		return 0, "", 0, fmt.Errorf("[parseTraining] неверный формат количества шагов %s: %w", strData[0], err)
```